### PR TITLE
Creating basic structure and adding basic github actions to the repo.

### PR DESCRIPTION
**Pull request by atorkmabrains**
The following has been updated in this PR:

- Updated basic folder structure.
- Added basic github actions

### CODE REVIEW

**Comment by atorkmabrains** [comment]
@proppy and @mithro This is my first of series of PRs. Please review and we will start migrating the rule deck to here.

File: .gitignore
Comment (mithro) [inline .gitignore:162]: We should just use the `.gitignore` for Python projects that GitHub has -> https://github.com/github/gitignore/blob/main/Python.gitignore

File: .github/workflows/regression.yml
Comment (mithro) [inline .github/workflows/regression.yml]: We *can* use the docker image like this but we probably can't depend on a docker image under the `mabrains` name. 

We should move it to either;
 * https:/github.com/hdl
 * Same place as https://github.com/google/skywater-pdk-actions/tree/main/run-drc-for-cell-gds-using-magic

File: AUTHORS
Comment (mithro) [inline AUTHORS:23]: Extra new line at end of file.

File: .github/workflows/linting.yml
Comment (mithro) [inline .github/workflows/linting.yml]: Should these be in a `requirements.txt` or `requirements.dev.txt` file? Then we can just `pip install -r requirements.txt`?

**Review by mithro — COMMENTED** [review]
Looking pretty good - just some minor comments.

File: .github/workflows/regression.yml
Comment (atorkmabrains) [reply .github/workflows/regression.yml]: Sounds great. We build the PDK images using our repo at: https://github.com/mabrains/build_mabrains_packages/tree/main 
HDL team can fork our repo and build it at their end. Or we could use a different way that would have the tools that we need.

Comment (proppy) [reply .github/workflows/regression.yml]: Is there a specific need to maintain another image?

There are already image for klayout and magic here:
https://hdl.github.io/containers/ToolsAndImages.html

Alternatively we could also use the conda package from:
https://anaconda.org/litex-hub/klayout
https://anaconda.org/litex-hub/magic

File: .github/workflows/linting.yml
Comment (proppy) [reply .github/workflows/linting.yml]: +1

File: requirements.txt
Comment (proppy) [inline requirements.txt]: is that actually used?

File: .github/workflows/regression.yml
Comment (proppy) [inline .github/workflows/regression.yml]: should we add a dummy Makefile, otherwise this will fail the CI.

**Review by proppy — APPROVED** [review]
Just some nits.

File: .gitignore
Comment (atorkmabrains) [reply .gitignore:162]: Resolved:
https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pv/pull/1/commits/5cd1182ba42cc63c3da4d924882b1128c7a5a151

File: .github/workflows/regression.yml
Comment (atorkmabrains) [reply .github/workflows/regression.yml]: @proppy I think that's a good starting point for our collaboration. I could do it if you want. But you asked me to add you to the development cycle. Do you want to do that?

I have given you access to the repo: https://github.com/efabless/globalfoundries-pdk-libs-gf180mcu_fd_pv so you would be able to make those changes.

Waiting for your comment on this.

File: .github/workflows/linting.yml
Comment (atorkmabrains) [reply .github/workflows/linting.yml]: @mithro and @proppy I could add that `requirements.txt`. But I wasn't sure what to do here as those only are used as part of the github actions only. I keep all the really required packages in `requirements.txt` file.

What's your recommendation here add it to `requirements.txt` or not? Keep in mind that those are only used for qualification purposes.

File: requirements.txt
Comment (atorkmabrains) [reply requirements.txt]: @proppy  It will be definitely used in the run scripts. But at that point, I just wanted to add something not to add an empty file for this PR. What's your recommendation here? Remove it and keep the file empty or just keep it as we know for sure it will be used and we are building the infrastructure?

File: .github/workflows/regression.yml
Comment (atorkmabrains) [reply .github/workflows/regression.yml]: @proppy I remembered, that we agreed the CI will always start with errors at the first few PRs and will be fixed and start to be clean as we go. I thought that was acceptable according to our agreement in the meeting. Do you want me to add a base make file in this PR as well?

**Comment by atorkmabrains** [comment]
@proppy Thanks for approving this. 

@proppy and @mithro What's next are you going to merge or wait for all open issues to be resolved before we merge?

File: .gitignore
Comment (atorkmabrains) [reply .gitignore:162]: @mithro I'll not `resolve conversation`. I want you to approve the changes.

File: AUTHORS
Comment (atorkmabrains) [reply AUTHORS:23]: Resolved at: https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pv/pull/1/commits/29982d051040d7d15a8070bef903eb32f77a12fd

File: .gitignore
Comment (mithro) [reply .gitignore:162]: LGTM.

File: AUTHORS
Comment (mithro) [reply AUTHORS:23]: LGTM.

File: .github/workflows/linting.yml
Comment (mithro) [reply .github/workflows/linting.yml]: What to put in your `requirements.txt` is confusing even for long time Python users.

My general guidance is that;
 * `requirements.txt` should be anything needed to setup the Python environment to run the tests successfully.
 * `requirements.dev.txt` should be used to install anything needed to do development. 

You can even include `requirements.txt` inside `requirements.dev.txt` with a `-r requirements.txt` as the first line in `requirements.txt` as shown at https://realpython.com/lessons/production-vs-development-dependencies/

If you are creating a Python package then there is also the `install_requires` in the `setup.py` file which describes what is need to run your Python package. There is some older guidance about this topic at https://caremad.io/posts/2013/07/setup-vs-requirement/ and some newer guidance at https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/

(Just FYI the full `requirements.txt` format is described at https://pip.pypa.io/en/latest/reference/requirements-file-format/#requirements-file-format)

If you have binary dependencies that are not available on PyPi then you need to use conda and an `environment.yml` file (see https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#create-env-file-manually), An `environment.yml` files is like a `requirements.txt` on steroids but also lets to specify conda packages. To make setting up a conda environment quick and easy, I created the `make-env` tool which can be found at https://github.com/SymbiFlow/make-env which will do all the work to download conda and setup the Python environment.

File: requirements.txt
Comment (proppy) [reply requirements.txt]: just adding a `# comment` should be fine but git should also accept empty files.

File: .github/workflows/regression.yml
Comment (proppy) [reply .github/workflows/regression.yml]: I'm also happy to land this PR as it, as long as we log an issue that we need to reconcile the container image (I'm also more than happy to help there!)

We should probably look at consolidating things with https://github.com/efabless/mpw_precheck/ images since it also need physical verification tool + pdks.

File: .github/workflows/regression.yml
Comment (proppy) [reply .github/workflows/regression.yml]: I think that's less critical to land the CI first now that we are starting from an empty repo, so I'd be in favor of include "some checks" to see that the CI is working.

@jeffdi might also have some input there.

File: .github/workflows/regression.yml
Comment (atorkmabrains) [reply .github/workflows/regression.yml]: https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pv/issues/2

@proppy as requested. Let's target that in a different PR.

File: pdk_regression.yml
Comment (proppy) [inline pdk_regression.yml]: `klayout` amd `magic` should be enough (unless you really want to pin everything)

File: Makefile
Comment (proppy) [inline Makefile]: `s/ym/yml/` ?

File: .github/workflows/linting.yml
Comment (proppy) [inline .github/workflows/linting.yml]: shoudl we choose a descriptive name here? (or remove it to use the default)

Comment (proppy) [reply .github/workflows/linting.yml]: oh nvm, I get now this is enabling the base environment
https://github.com/marketplace/actions/setup-miniconda#activate-base-environment

maybe we should add a comment?

File: .github/workflows/regression.yml
Comment (atorkmabrains) [reply .github/workflows/regression.yml]: @proppy and @mithro Completed the migration to use conda environment using make-env git repo.

https://github.com/efabless/globalfoundries-pdk-libs-gf180mcu_fd_pv/commit/19cb6f2ac68e59fd009b713a170bceee44487daa

Please review and resolve if you agree with the changes.

File: .github/workflows/linting.yml
Comment (atorkmabrains) [reply .github/workflows/linting.yml]: @mithro and @proppy Same as above: 
https://github.com/efabless/globalfoundries-pdk-libs-gf180mcu_fd_pv/commit/19cb6f2ac68e59fd009b713a170bceee44487daa

File: requirements.txt
Comment (atorkmabrains) [reply requirements.txt]: @proppy removed and now using the `make-env` setup.


https://github.com/efabless/globalfoundries-pdk-libs-gf180mcu_fd_pv/commit/19cb6f2ac68e59fd009b713a170bceee44487daa

File: .github/workflows/regression.yml
Comment (atorkmabrains) [reply .github/workflows/regression.yml]: @proppy I have changed the code and added the `makefile` in the main part as shown in the last commit. I have added empty targets for now. I'll change that later.

File: pdk_regression.yml
Comment (atorkmabrains) [reply pdk_regression.yml]: Done @proppy https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pv/pull/1/commits/d55d4d845918501afadda47d724e606d3e06f0c4

File: Makefile
Comment (atorkmabrains) [reply Makefile]: @proppy Fixed and now passing.

File: .github/workflows/linting.yml
Comment (atorkmabrains) [reply .github/workflows/linting.yml]: @proppy removed. No longer needed.

**Comment by atorkmabrains** [comment]
@proppy and @mithro I don't have merge rights on this repo. You guys need to merge this once you approve it.

File: pdk_regression.yml
Comment (proppy) [inline pdk_regression.yml:8]: maybe also add `- python` before `pip` to be a little bit more explicit and be sure we don't default on the python from the parent environment (being host or conda)

Comment (atorkmabrains) [reply pdk_regression.yml:8]: Done https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pv/pull/1/commits/1728f6344781d4575c648aa932bab53a77ce07e4

@proppy

File: .github/workflows/linting.yml
Comment (proppy) [inline .github/workflows/linting.yml:18]: add workflow_dispatch so we can test manually?

File: .github/workflows/regression.yml
Comment (proppy) [inline .github/workflows/regression.yml:26]: add push?

**Review by proppy — APPROVED** [review]

File: pdk_regression.yml
Comment (proppy) [inline pdk_regression.yml:8]: I don't think you need the duplicate `pip` here just:
```
- python
- pip:
  - requirements.txt
```

File: .github/workflows/linting.yml
Comment (atorkmabrains) [reply .github/workflows/linting.yml:18]: Done https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pv/pull/1/commits/b450e352c1dbdda8c9c579389c6e6eed4532cb34

@proppy

File: .github/workflows/regression.yml
Comment (atorkmabrains) [reply .github/workflows/regression.yml:26]: I intentionally removed originally. Because DRC regression usually takes very long time to run. Anyhow, I have added it back.

https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pv/pull/1/commits/b450e352c1dbdda8c9c579389c6e6eed4532cb34

Done

@proppy

File: pdk_regression.yml
Comment (atorkmabrains) [reply pdk_regression.yml:8]: @proppy I followed the example: https://github.com/SymbiFlow/make-env/blob/33b80bd32c30fb8affd0fd5cda544d1bca075593/test/environment.yml#L14

Are you sure it will work if I removed it?

File: .github/workflows/regression.yml
Comment (proppy) [inline .github/workflows/regression.yml]: Looks like this currently fail:
https://github.com/efabless/globalfoundries-pdk-libs-gf180mcu_fd_pv/actions/runs/3949971157/jobs/6761875635

Should `>` be `|` ?

File: .github/workflows/regression.yml
Comment (proppy) [inline .github/workflows/regression.yml]: you should be able add:
```
with:
  submodules: 'recursive'
```
to `actions/checkout@v3` to get the same effect, see https://github.com/actions/checkout#usage

File: .github/workflows/linting.yml
Comment (proppy) [inline .github/workflows/linting.yml]: you should be able add:
```
with:
  submodules: 'recursive'
```
to `actions/checkout@v3` to get the same effect, see https://github.com/actions/checkout#usage

File: pdk_regression.yml
Comment (proppy) [reply pdk_regression.yml:8]: ah my bad: you're right, that's what the conda docs have too:
https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-file-manually

**Review by proppy — CHANGES_REQUESTED** [review]
should we get the `regression` workflow passing before merging?
https://github.com/efabless/globalfoundries-pdk-libs-gf180mcu_fd_pv/actions/runs/3949971157

**Comment by atorkmabrains** [comment]
@proppy Agree with you. I'm stuck today in GSA Egypt Summit.

You could give it a hit at your end and fix it. Or wait for me to fix but that will be on Sunday. It would be nice to see you contribute to the work as you requested.

**Comment by atorkmabrains** [comment]
@proppy No pressure, you could do it or wait for me.

**Comment by proppy** [comment]
@atorkmabrains sure, after you review and merge https://github.com/efabless/globalfoundries-pdk-libs-gf180mcu_fd_pv/pull/1 in the fork, the commit should show up in this PR.

**Comment by atorkmabrains** [comment]
@proppy Done. Thanks for all the help.

@mithro Could you please merge?

**Comment by proppy** [comment]
For reference the CI now passes:
https://github.com/efabless/globalfoundries-pdk-libs-gf180mcu_fd_pv/actions

**Review by proppy — APPROVED** [review]

**Comment by atorkmabrains** [comment]
I'm enjoying this @proppy Happy to be working with you.

File: .github/workflows/regression.yml
Comment (atorkmabrains) [reply .github/workflows/regression.yml]: @proppy Could you please resolve this?

File: .github/workflows/regression.yml
Comment (atorkmabrains) [reply .github/workflows/regression.yml]: @proppy Also this?

File: .github/workflows/linting.yml
Comment (atorkmabrains) [reply .github/workflows/linting.yml]: @proppy Also this?

**Comment by proppy** [comment]
@atorkmabrains yep, I enjoyed it too :)

The PR is really minimal and clean, great work on this !

**Comment by atorkmabrains** [comment]
@mithro Could you please give @proppy merge rights?

**Comment by mithro** [comment]
From what I can see, @proppy should already have merge rights?

**Comment by atorkmabrains** [comment]
@proppy Could you please confirm you have merge rights?

@mithro I have another request here, I believe you are busy and @proppy might have more time to help us with the merge process. Could he be the main reviewer for this?

**Comment by mithro** [comment]
100% - @proppy is the lead/primary reviewer. I'll add things when I can but if @proppy is happy then he should merge away.

**Comment by mithro** [comment]
IE Don't wait / block on my feedback / review unless @proppy has a very specific thing he wants me to look at.

**Comment by atorkmabrains** [comment]
Thanks @mithro.

@proppy I'll make you as the main reviewer.

**Pull request merged**
